yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -318,6 +314,13 @@ private[spark] class Client(
     env("SPARK_YARN_STAGING_DIR") = stagingDir
     env("SPARK_USER") = UserGroupInformation.getCurrentUser().getShortUserName()
 
+    // Propagate SPARK_HOME to the containers. This is needed for pyspark to
+    // work, since the executor's PYTHONPATH is built based on the location
+    // of SPARK_HOME.
+    sparkConf.getOption("spark.home").orElse(sys.env.get("SPARK_HOME")).foreach { path =>
+      env("SPARK_HOME") = path
+    }
```

Comment:
It's not entirely related to the patch. But without this I just could not run pyspark apps on yarn as part of testing. I couldn't for the life of me find out how SPARK_HOME propagated to executors before.